corehq/apps/case_search/models.py

```diff
@@ -0,0 +1,118 @@
+from django.db import models
+from jsonfield.fields import JSONField
+
+FUZZY_PROPERTIES = "fuzzy_properties"
+
+
+class CaseSearchConfig(models.Model):
+    """
+    Contains config for case search
+    """
+    class Meta:
+        app_label = 'case_search'
+
+    domain = models.CharField(
+        max_length=256,
+        null=False,
+        blank=False,
+        db_index=True,
+        primary_key=True
+    )
+    enabled = models.BooleanField(blank=False, null=False, default=False)
+    config = JSONField(default={})
+
+    @classmethod
+    def enabled_domains(cls):
+        return cls.objects.filter(enabled=True).values_list('domain', flat=True)
+
+    def add_fuzzy_property(self, case_type, property):
+        """
+        Adds a case property to be fuzzy searched with CaseSearchES
+
+        Case Properties set as fuzzy will use a fuzzy search flag in ES.
+        Should only really be set for text properties.
+        Fuzzy properties add the following to the config JSON:
+        {
+            "fuzzy_properties":[
+                {
+                    "case_type": "pirates",
+                    "properties": ["name", "age"]
+                },
+                {
+                    "case_type": "swashbucklers",
+                    "properties": ["has_parrot"]
+                }
+            ]
+        }
+        """
```

Comment:
I think it'd be nice if `self.config` was a `JsonObject`. Then we could clearly define the schema we're expecting and push operations on the config into the config object itself. You could trivially do this by just adding a `wrapped_config` property to the model, or (out of scope) you could do something fancy like make a `JsonObjectField` subclass of `JsonField`.